File: src/main/java/Franchise.java
public class Franchise {

    private final Restaurant[] _restaurants;

    public Franchise(Restaurant[] restaurants) {
        _restaurants = restaurants;
    }

    public float ChiffreAffaires() {
        float total = 0;

        for (Restaurant r : _restaurants) {
            total += r.ChiffreAffaires();
        }

        return total;
    }


}

File: src/app/LeGrandRestaurant_AvecPatterns/LeGrandRestaurant/Franchise.cs
using System.Linq;

namespace LeGrandRestaurant
{
    public class Franchise
    {
        private readonly Restaurant[] _restaurants;

        public Franchise(params Restaurant[] restaurants)
        {
            _restaurants = restaurants;
        }

        public decimal ChiffreAffaires
            => _restaurants.Sum(restaurant => restaurant.ChiffreAffaires);
    }
}
